File: android/app/src/main/java/com/timaurora/tls/callback/PwdLoginCb.java
package com.timaurora.tls.callback;

import com.facebook.react.bridge.Arguments;
import com.facebook.react.bridge.Promise;
import com.facebook.react.bridge.WritableMap;
import com.timaurora.tim.PromiseHelper;
import com.timaurora.tls.AuthHelper;

import tencent.tls.platform.TLSErrInfo;
import tencent.tls.platform.TLSPwdLoginListener;
import tencent.tls.platform.TLSUserInfo;

public class PwdLoginCb implements TLSPwdLoginListener {
    private String name;
    private Promise promise;
    public PwdLoginCb(String tag, Promise promise){
        this.name = tag;
        this.promise = promise;
    }
    @Override
    public void OnPwdLoginSuccess(TLSUserInfo tlsUserInfo) {
        String userSig = AuthHelper.tlsHelper.getUserSig(tlsUserInfo.identifier);
        String msg = name + "succ. userSig:" + userSig;
        WritableMap map = Arguments.createMap();
        map.putString("msg",msg);
        map.putInt("code",200);
        map.putString("userSig",userSig);
        promise.resolve(map);
    }

    @Override
    public void OnPwdLoginReaskImgcodeSuccess(byte[] bytes) {
        /* 请求刷新图片验证码成功，此时需要用picData 更新验证码图片，原先的验证码已经失效*/
        String msg = name + "succ ";
        promise.resolve(PromiseHelper.createSucResolve(msg));
    }

    @Override
    public void OnPwdLoginNeedImgcode(byte[] bytes, TLSErrInfo tlsErrInfo) {
        //登录需要验证码
        String msg = name + " need imgcode";
        WritableMap map = Arguments.createMap();
        map.putString("msg",msg);
        map.putInt("code",201);
        promise.resolve(map);
    }

    @Override
    public void OnPwdLoginFail(TLSErrInfo tlsErrInfo) {
        String msg = name + " fail. code:" + tlsErrInfo.ErrCode + ",msg:" + tlsErrInfo.Msg;
        promise.reject(PromiseHelper.createErrReject(msg));
    }

    @Override
    public void OnPwdLoginTimeout(TLSErrInfo tlsErrInfo) {
        String msg = name + " fail. code:" + tlsErrInfo.ErrCode + ",msg:" + tlsErrInfo.Msg;
        promise.reject(PromiseHelper.createErrReject(msg));
    }
}

File: ios/RCTLS/Callback/PwdLoginCb.m
//
// Created by 美德远健 on 2018/7/30.
// Copyright (c) 2018 Facebook. All rights reserved.
//

#import <TLSSDK/TLSHelper.h>
#import "PwdLoginCb.h"


@implementation PwdLoginCb {

}
- (instancetype)initWithName:(NSString *)name resolve:(RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject {
    self.name = name;
    self.resolve = resolve;
    self.reject = reject;
    return self;
}

- (void)OnPwdLoginNeedImgcode:(NSData *)picData andErrInfo:(TLSErrInfo *)errInfo {
    [self rejectWithTLSErrInfo:errInfo];
}

- (void)OnPwdLoginReaskImgcodeSuccess:(NSData *)picData {
    NSString *msg = [[NSString alloc] initWithFormat:@"%@ succ.",_name];
    _resolve([PromiseUtil createSuccResolve:msg]);
}

- (void)OnPwdLoginSuccess:(TLSUserInfo *)userInfo {
    NSString *msg = [[NSString alloc] initWithFormat:@"%@ succ.",_name];
    NSString *userSig = [[TLSHelper getInstance] getTLSUserSig:[userInfo identifier]];
    NSDictionary *dict = [PromiseUtil createSuccResolve:msg];
    [dict setValue:userSig forKey:@"userSig"];
    _resolve(dict);
}

- (void)OnPwdLoginFail:(TLSErrInfo *)errInfo {
    [self rejectWithTLSErrInfo:errInfo];
}

- (void)OnPwdLoginTimeout:(TLSErrInfo *)errInfo {
    [self rejectWithTLSErrInfo:errInfo];
}

- (void)rejectWithTLSErrInfo:(TLSErrInfo *)errInfo{
    NSString *msg = [[NSString alloc] initWithFormat:@"%@ fail. code:%d ,msg:%@",_name, [errInfo dwErrorCode], [errInfo sErrorMsg]];
    [PromiseUtil rejectWithMsg:_reject msg:msg];
}
@end